applications/crossbar/src/crossbar_doc.erl

```diff
@@ -395,33 +398,34 @@ view_sort_direction(Options) ->
         'undefined' -> 'ascending'
     end.
 
-load_view(#load_view_params{dbs=[]
-                           ,context=Context
+load_view(#load_view_params{dbs = []
+                           ,context = Context
                            }) ->
     case cb_context:resp_status(Context) of
         'success' ->
             lager:debug("databases exhausted"),
             handle_datamgr_success(cb_context:doc(Context), Context);
         _Status -> Context
     end;
-load_view(#load_view_params{page_size=PageSize
-                           ,context=Context
-                           }) when PageSize =< 0 ->
+load_view(#load_view_params{page_size = PageSize
+                           ,context = Context
+                           ,should_paginate = true
+                           })
+  when is_integer(PageSize), PageSize =< 0 ->
     lager:debug("page_size exhausted: ~p", [PageSize]),
-    case cb_context:resp_status(Context) of
-        'success' -> handle_datamgr_success(cb_context:doc(Context), Context);
-        _Status -> Context
+    case success =:= cb_context:resp_status(Context) of
```

Comment:
yes :D
no this was caught in the middle of debugging and it looks like this after reversing some of my changes here